### ISSUE TITLE
ElevatedButton onPressed not working inside nested Stack

### ISSUE THREAD

**Issue by FlaviusBanc** [flutter/flutter#113074]
My buttons are not being detected when I tap on them

**Note:** I have an animation that somewhat hides the button from the screen (the button is not visible when animation is played)

Stack -> Stack -> ElevatedButton (simplified)

I have a header for the application that's just inside the first Stack, the button inside the first stack is working perfectly but the other buttons inside the second Stack are not being detected at all.

**Sample code project structure**
![image](https://user-images.githubusercontent.com/109205841/194429744-f652d8a8-9b65-4ad0-80bf-4532b30d9495.png)

Build method for the home page

```
@override
  Widget build(BuildContext context) {
    return Material(
      child: GestureDetector(
        onHorizontalDragStart: _onDragStart,
        onHorizontalDragUpdate: _onDragUpdate,
        onHorizontalDragEnd: _onDragEnd,
        child: Stack(
          clipBehavior: Clip.none,
          children: <Widget>[
            Container(color: COLOR_BLUE_ACCENT),
            _buildBackground(),
            _buildHeader(),
          ],
        ),
      ),
    );
  }
```

The following code is the display for the home page, the ElevatedButton is being created inside the addButton() method
The created button is never accessed, there might be something on top of it, but I have not managed to find the problem.

```
_buildBackground() {
    return Positioned.fill(
      top: -_extraHeight,
      bottom: -_extraHeight,
      child: AnimatedBuilder(
        animation: _animator,
        builder: (context, widget) => Transform.translate(
          offset: Offset(_maxSlide * _animator.value, 0),
          child: Transform(
            transform: Matrix4.identity()
              ..setEntry(3, 2, 0.001)
              ..rotateY((pi / 2 + 0.1) * -_animator.value),
            alignment: Alignment.centerLeft,
            child: widget,
          ),
        ),
        child: Container(
          color: COLOR_LIGHT_BACKGROUND,
          width: _screen.width,
          height: _screen.height,
          child: Stack(
            children: [
              Positioned(
                  width: _screen.width,
                  top: _extraHeight + 0.1 * _screen.height,
                  child: todayPage()),
              AnimatedBuilder(
                animation: _animator,
                builder: (_, __) => Container(
                  color: Colors.black.withAlpha(
                    (150 * _animator.value).floor(),
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget todayPage() {
    return SingleChildScrollView(
      physics: BouncingScrollPhysics(),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.center,
        children: [
          addDivider(),
          addVerticalSpace(20),
          addButton(BUTTON_ADD_TASK, Icons.add, 130.0),
        ],
      ),
    );
  }

  Widget addButton(btnText, icon, width) {
    return Container(
      width: width,
      child: ElevatedButton(
        style: ElevatedButton.styleFrom(
          shape:
          RoundedRectangleBorder(borderRadius: BorderRadius.circular(20.0)),
          backgroundColor: COLOR_BLUE_ACCENT, // background (button) color
        ),
        onPressed: () => print('addbutton'), //THIS AIN'T WORKING WHY PLEASE GOD WHYYYY
        child: Row(
          children: [
            Icon(icon),
            addHorizontalSpace(4.5),
            Text(btnText),
          ],
        ),
      ),
    );
  }
```

**FULL SAMPLE CODE**

**home_page.dart**
```import 'dart:math';
import 'package:flutter/material.dart';

const COLOR_BLUE_ACCENT = Color.fromRGBO(0, 90, 168, 1);
const COLOR_LIGHT_BACKGROUND = Color.fromRGBO(211, 211, 211, 1);
const COLOR_GREY = Color.fromRGBO(41, 41, 41, 1);

const BUTTON_ADD_TASK = "ADD TASK";

String _mainPageTitle = "TO-DO TODAY";

class HomePage extends StatefulWidget {
  final Widget child;

  HomePage({required this.child});

  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage>
    with SingleTickerProviderStateMixin {
  var _maxSlide = 0.75;
  var _extraHeight = 0.1;
  late double _startingPos;
  var _drawerVisible = false;
  late AnimationController _animationController;
  Size _screen = Size(0, 0);
  late CurvedAnimation _animator;

  @override
  void initState() {
    super.initState();
    _animationController = AnimationController(
      vsync: this,
      duration: const Duration(milliseconds: 800),
    );
    _animator = CurvedAnimation(
      parent: _animationController,
      curve: Curves.easeInOutQuad,
      reverseCurve: Curves.easeInQuad,
    );
  }

  @override
  void didChangeDependencies() {
    _screen = MediaQuery.of(context).size;
    _maxSlide *= _screen.width;
    _extraHeight *= _screen.height;
    super.didChangeDependencies();
  }

  @override
  Widget build(BuildContext context) {
    return Material(
      child: GestureDetector(
        onHorizontalDragStart: _onDragStart,
        onHorizontalDragUpdate: _onDragUpdate,
        onHorizontalDragEnd: _onDragEnd,
        child: Stack(
          clipBehavior: Clip.none,
          children: <Widget>[
            Container(color: COLOR_BLUE_ACCENT),
            _buildBackground(),
            _buildHeader(),
          ],
        ),
      ),
    );
  }

  void _onDragStart(DragStartDetails details) {
    _startingPos = details.globalPosition.dx;
  }

  void _onDragUpdate(DragUpdateDetails details) {
    final globalDelta = details.globalPosition.dx - _startingPos;
    if (globalDelta > 0) {
      final pos = globalDelta / _screen.width;
      if (_drawerVisible && pos <= 1.0) return;
      _animationController.value = pos;
    } else {
      final pos = 1 - (globalDelta.abs() / _screen.width);
      if (!_drawerVisible && pos >= 0.0) return;
      _animationController.value = pos;
    }
  }

  void _onDragEnd(DragEndDetails details) {
    if (details.velocity.pixelsPerSecond.dx.abs() > 500) {
      if (details.velocity.pixelsPerSecond.dx > 0) {
        _animationController.forward(from: _animationController.value);
        _drawerVisible = true;
      } else {
        _animationController.reverse(from: _animationController.value);
        _drawerVisible = false;
      }
      return;
    }
    if (_animationController.value > 0.5) {
      {
        _animationController.forward(from: _animationController.value);
        _drawerVisible = true;
      }
    } else {
      {
        _animationController.reverse(from: _animationController.value);
        _drawerVisible = false;
      }
    }
  }

  void _toggleDrawer() {
    if (_animationController.value < 0.5)
      _animationController.forward();
    else
      _animationController.reverse();
  }

  _buildBackground() {
    return Positioned.fill(
      top: -_extraHeight,
      bottom: -_extraHeight,
      child: AnimatedBuilder(
        animation: _animator,
        builder: (context, widget) => Transform.translate(
          offset: Offset(_maxSlide * _animator.value, 0),
          child: Transform(
            transform: Matrix4.identity()
              ..setEntry(3, 2, 0.001)
              ..rotateY((pi / 2 + 0.1) * -_animator.value),
            alignment: Alignment.centerLeft,
            child: widget,
          ),
        ),
        child: Container(
          color: COLOR_LIGHT_BACKGROUND,
          width: _screen.width,
          height: _screen.height,
          child: Stack(
            children: [
              Positioned(
                  width: _screen.width,
                  top: _extraHeight + 0.1 * _screen.height,
                  child: todayPage()),
              AnimatedBuilder(
                animation: _animator,
                builder: (_, __) => Container(
                  color: Colors.black.withAlpha(
                    (150 * _animator.value).floor(),
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget todayPage() {
    return SingleChildScrollView(
      physics: BouncingScrollPhysics(),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.center,
        children: [
          addDivider(),
          addVerticalSpace(20),
          addButton(BUTTON_ADD_TASK, Icons.add, 130.0),
        ],
      ),
    );
  }

  Widget addButton(btnText, icon, width) {
    return Container(
      width: width,
      child: ElevatedButton(
        style: ElevatedButton.styleFrom(
          shape:
          RoundedRectangleBorder(borderRadius: BorderRadius.circular(20.0)),
          backgroundColor: COLOR_BLUE_ACCENT, // background (button) color
        ),
        onPressed: () => print('addbutton'),
        child: Row(
          children: [
            Icon(icon),
            addHorizontalSpace(4.5),
            Text(btnText),
          ],
        ),
      ),
    );
  }

  _buildHeader() => SafeArea(
        child: AnimatedBuilder(
            animation: _animator,
            builder: (_, __) {
              return Transform.translate(
                offset: Offset((_screen.width - 60) * _animator.value, 0),
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: <Widget>[
                    SizedBox(
                      width: 50,
                      height: 50,
                      child: InkWell(
                        onTap: _toggleDrawer,
                        child: Icon(Icons.menu_rounded, color: COLOR_GREY),
                      ),
                    ),
                    Opacity(
                      opacity: 1 - _animator.value,
                      child: Text(
                        _mainPageTitle,
                        style: TextStyle(
                            fontWeight: FontWeight.w900,
                            color: COLOR_BLUE_ACCENT),
                      ),
                    ),
                    addHorizontalSpace(50.0),
                  ],
                ),
              );
            }),
      );

  Widget addDivider() {
    return const Divider(
      color: COLOR_GREY,
      height: 10,
      thickness: 1,
      indent: 10,
      endIndent: 10,
    );
  }

  Widget addVerticalSpace(double height) {
    return SizedBox(height: height);
  }

  Widget addHorizontalSpace(double width) {
    return SizedBox(width: width);
  }
}
```

**main.dart**
```
import 'package:flutter/material.dart';
import 'home_page.dart';

void main() => runApp(MyApp());


class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) => MaterialApp(
        title: 'Agenda',
        color: Color.fromRGBO(41, 41, 41, 1),
        home: HomePage(child: this),
      );
}
```

Thank you in advance for your time.
It might be an obvious fix, but I scrapped the internet for an answer and couldn't find any solution. :)

**Comment by huycozy** [comment]
Hi @FlaviusBanc,  
For the issue to be workable, it needs to be reproducible with a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that doesn’t include 3rd party plugins or complex production code. Please provide such a sample. 

Also, please provide the output of `flutter doctor -v` as well. Thank you!

**Comment by FlaviusBanc** [comment]
I have managed to fixed the problem.

Context: I had created the AnimationBuilder after creating the Widgets for the buttons, so the animation was in front of the buttons that's why the onPressed buttons weren't detected.

Fix: I had changed the order in which I created the widgets
old way - Stack { children: ElevatedButton, AnimationBuilder }
new way - Stack { children: AnimationBuilder, ElevatedButton }

I don't know how I missed this but after a few hours of chillin I had come to this realisation.

Hopefully it's understandable. :)
